### ISSUE TITLE
Mafia 1 - strange reverb feedback loop in some locations

### ISSUE THREAD

**Issue by Lrflx** [kcat/dsoal#94]
During my last playthrough I noticed that in several locations something strange definitely was happening. Easiest to check is in prison tower, with alchemy there's a slight humming trail going on for couple seconds after every gunshot. And with dsoal it just keeps going on and on instead, practically indifinetiley and doesn't stop even in main menu. Tested it with reverb boost setting and it does change reverb in general, but doesn't have any audible effect on that feedback loop humming.
[Mafia.zip](https://github.com/kcat/dsoal/files/12557876/Mafia.zip)

**Comment by mirh** [comment]
With the [last git](https://github.com/ThreeDeeJay/dsoal/releases)?

**Comment by kcat** [comment]
FWIW, there is newer code in the [C++ rewrite branch](https://github.com/kcat/dsoal/tree/c++-rewrite) (although I don't expect EAX/reverb behavior to be much different; and that branch is currently missing a few features like capture/duplex operation and WRITEPRIMARY support, that still need to be reimplemented).

I do need to find some way to log the EAX parameters being set, in a way that doesn't significantly impact performance when not logging. The reverb gain shouldn't be significantly different, nor should the decay tail, so it'd be helpful to see what EAX property values are being set.

**Comment by Lrflx** [comment]
Ok, I'll be glad to retest it whenever it comes out. To be honest I'm not sure if it's even a reverb, sounds more like reflections, don't know for sure if EAX 3 can to it.

**Comment by kcat** [comment]
Are you able to provide a recording of it happening? Preferably of both ALchemy and DSOAL for comparison.

**Comment by Lrflx** [comment]
Of course. Some additional clipping got there during capture, it wasn't in game originally
[Mafia.Alchemy.DSOAL.zip](https://github.com/kcat/dsoal/files/12562795/Mafia.Alchemy.DSOAL.zip)

**Comment by princhonzik** [comment]
Compiled the C++ rewrite + latest OpenAL Soft and the issue is still there. It actually is reverb because cancelling it by setting the boost value low also affects the effect.

It's not in just some locations, however, but most of them, only in most the effect is quiet enough compared to the rest of the sounds not to be very disturbing. Load anything, pause the game (stops all other sounds), increase speaker volume, the ominous hum loop is there. If you play longer, some of the effects layer up (more hum), some are terminated when you leave the location [no loads in-between]). The loop continues even when loading the main menu (which is just another game scene), and only resets if you load a different scene.

If there's anything to test in the future, I'll be glad to do so.

**Comment by kcat** [comment]
The latest Git build of OpenAL Soft includes an option to trace the reverb parameters being set (either set the `ALSOFT_EAX_TRACE_COMMITS` environment variable to `1` or `true`, or set
```
[eax]
trace-commits = true
```
in alsoft.ini). Then a trace log of OpenAL Soft should include the reverb parameters that get set. If possible, updated OpenAL Soft and DSOAL logs would help.
